### PR TITLE
Use upstream `kubent` container image

### PR DESCRIPTION
**Pull request by simu**
There's now official `kubent` images available from https://github.com/doitintl/kube-no-trouble/pkgs/container/kube-no-trouble.

This commit switches the component template to use that image instead of the one-off image we built ourselves.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: .github/workflows/sync_base.yaml [hunk 1]
```diff
@@ -18,13 +18,14 @@ jobs:
...
+        id: setup-python
         with:
           python-version: '3.10'
       - uses: actions/cache@v3
         with:
           path: |
             ~/.commodore-venv/
-          key: ${{ runner.os }}-commodore-${{ hashFiles('requirements.txt') }}
+          key: ${{ runner.os }}-commodore-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('requirements.txt') }}
       - name: Install Commodore
```
Comment (glrf) [inline .github/workflows/sync_base.yaml:29]: This is a good idea, but seems unrelated

**Review by glrf — APPROVED** [review]
I'm a bit torn if we should really use the latest image.

Having a version pin creates noise on every component if we update it, but if latest ever breaks this is also inconvenient. I'm ok with both approaches

File: .github/workflows/sync_base.yaml (hunk 1 above)
Comment (simu) [reply .github/workflows/sync_base.yaml:29]: It's unrelated, except that the tests were breaking without it.

**Comment by simu** [comment]
> I'm a bit torn if we should really use the latest image.

We're using `latest` for all other images used in the make targets, so I feel like it makes the most sense to use `latest` here to keep it consistent.

**Comment by glrf** [comment]
> We're using `latest` for all other images used in the make targets, so I feel like it makes the most sense to use `latest` here to keep it consistent.

Fair enough, I thought we pinned external images

File: .github/workflows/sync_base.yaml (hunk 1 above)
Comment (glrf) [reply .github/workflows/sync_base.yaml:29]: Makes sense

Comment (simu) [reply .github/workflows/sync_base.yaml:29]: factored out into #25
